File: cpp/tests/SurroundedRegionsTest.cpp
#include "SurroundedRegions.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(SurroundedRegions, Solution130_1) {
  auto S130_1 = Solution130_1();

  std::vector<std::vector<char>> Board1{{'X', 'X', 'X', 'X'},
                                        {'X', 'O', 'O', 'X'},
                                        {'X', 'X', 'O', 'X'},
                                        {'X', 'O', 'X', 'X'}};
  std::vector<std::vector<char>> const Expected1{{'X', 'X', 'X', 'X'},
                                                 {'X', 'X', 'X', 'X'},
                                                 {'X', 'X', 'X', 'X'},
                                                 {'X', 'O', 'X', 'X'}};
  S130_1.solve(Board1);
  EXPECT_EQ(Expected1, Board1);

  std::vector<std::vector<char>> Board2{{'X'}};
  std::vector<std::vector<char>> const Expected2{{'X'}};
  S130_1.solve(Board2);
  EXPECT_EQ(Expected2, Board2);
}

File: java/src/test/java/io/binac/leetcode/SurroundedRegionsTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SurroundedRegionsTest {
    private static final SurroundedRegions.Solution1 SOLUTION1 = new SurroundedRegions.Solution1();

    private char[][] getBOARD1() {
        return new char[][]{{'X', 'X', 'X', 'X'},
                {'X', 'O', 'O', 'X'},
                {'X', 'X', 'O', 'X'},
                {'X', 'O', 'X', 'X'}};
    }

    private final char[][] EXPECTED1 = {
            {'X', 'X', 'X', 'X'},
            {'X', 'X', 'X', 'X'},
            {'X', 'X', 'X', 'X'},
            {'X', 'O', 'X', 'X'}
    };

    private char[][] getBOARD2() {
        return new char[][]{{'X'}};
    }

    private final char[][] EXPECTED2 = {{'X'}};

    @Test
    void test1() {
        char[][] board1 = getBOARD1();
        SOLUTION1.solve(board1);
        assertArrayEquals(EXPECTED1, board1);

        char[][] board2 = getBOARD2();
        SOLUTION1.solve(board2);
        assertArrayEquals(EXPECTED2, board2);
    }
}